`playaud.c: error: passing argument 4 of 'waveOutOpen' makes pointer from integer without a cast [-Wint-conversion]`

```
ccache /opt/local/bin/gcc-mp-14 -c -MMD -g -O2 -Wall -Wno-missing-braces -Wno-pointer-sign -Wno-unused-variable -Wno-unused-but-set-variable -Wno-unused-function -Wno-incompatible-pointer-types -I../include -I../api -I../cmd -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_audio_dectalk/dectalk/work/dectalk-2023-10-30/src/dapi/build/dic/10.0.0d2/gr/release -I../kernel -I../lts -I../nt -I../osf -I../ph -I../../../ -I../protos -I../vtm -I. -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_audio_dectalk/dectalk/work/dectalk-2023-10-30/src/dapi/build/dectalk//gr/release/include  -DLTSSIM -D_APPLE_MAC_ -D_BIGENDIAN_ -DTTSSIM -DANSI -DBLD_DECTALK_DLL -DGERMAN -DDECTALK_INSTALL_PREFIX=\"/opt/local\" -DACCESS32 -DTYPING_MODE -dynamic  -DUSE_PULSEAUDIO -D_REENTRANT -o ../../build/dectalk/10.0.0d2/gr/release/link/pipe.o pipe.c
ccache /opt/local/bin/gcc-mp-14 -c -MMD -g -O2 -Wall -Wno-missing-braces -Wno-pointer-sign -Wno-unused-variable -Wno-unused-but-set-variable -Wno-unused-function -Wno-incompatible-pointer-types -I../include -I../api -I../cmd -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_audio_dectalk/dectalk/work/dectalk-2023-10-30/src/dapi/build/dic/10.0.0d2/gr/release -I../kernel -I../lts -I../nt -I../osf -I../ph -I../../../ -I../protos -I../vtm -I. -I/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_audio_dectalk/dectalk/work/dectalk-2023-10-30/src/dapi/build/dectalk//gr/release/include  -DLTSSIM -D_APPLE_MAC_ -D_BIGENDIAN_ -DTTSSIM -DANSI -DBLD_DECTALK_DLL -DGERMAN -DDECTALK_INSTALL_PREFIX=\"/opt/local\" -DACCESS32 -DTYPING_MODE -dynamic  -DUSE_PULSEAUDIO -D_REENTRANT -o ../../build/dectalk/10.0.0d2/gr/release/link/mmalloc.o mmalloc.c
playaud.c: In function 'OpenWaveOutputDevice':
playaud.c:5937:29: error: passing argument 4 of 'waveOutOpen' makes pointer from integer without a cast [-Wint-conversion]
 5937 |                             (DWORD)WaveOutCallbackRoutine,
      |                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                             |
      |                             unsigned int
playaud.c:275:12: note: in definition of macro 'LOCK_MME_CALL'
  275 |            statement; \
      |            ^~~~~~~~~
In file included from playaud.c:241:
linux_audio.h:443:67: note: expected 'void (*)(void *, unsigned int,  unsigned int,  long int,  long int)' but argument is of type 'unsigned int'
  443 |                             const LPWAVEFORMATEX lpFormat, void (*dwCallback)(void *, unsigned int,  unsigned int,  long,  long),
      |                                                            ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
playaud.c:6000:31: error: passing argument 4 of 'waveOutOpen' makes pointer from integer without a cast [-Wint-conversion]
 6000 |                               (DWORD)WaveOutCallbackRoutine,
      |                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                               |
      |                               unsigned int
playaud.c:275:12: note: in definition of macro 'LOCK_MME_CALL'
  275 |            statement; \
      |            ^~~~~~~~~
linux_audio.h:443:67: note: expected 'void (*)(void *, unsigned int,  unsigned int,  long int,  long int)' but argument is of type 'unsigned int'
  443 |                             const LPWAVEFORMATEX lpFormat, void (*dwCallback)(void *, unsigned int,  unsigned int,  long,  long),
      |                                                            ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[3]: *** [../../build/dectalk/10.0.0d2/gr/release/link/playaud.o] Error 1
make[3]: *** Waiting for unfinished jobs....
```

--- a/src/dapi/src/nt/playaud.c
+++ b/src/dapi/src/nt/playaud.c
@@ -5934,7 +5934,7 @@ static MMRESULT OpenWaveOutputDevice( HPLAY_AUDIO_T pPlayAudio )
     mmStatus = waveOutOpen( &(pPlayAudio->hWaveOut),
                             pPlayAudio->uiSelectedDeviceID,
                             (LPWAVEFORMATEX)pWaveFormat,
-                            (DWORD)WaveOutCallbackRoutine,
+                            WaveOutCallbackRoutine,
                             (DWORD)pPlayAudio,
                             dwOpenFlags ))
 #endif
@@ -5997,7 +5997,7 @@ static MMRESULT OpenWaveOutputDevice( HPLAY_AUDIO_T pPlayAudio )
       mmStatus = waveOutOpen( &(pPlayAudio->hWaveOut),
                               pPlayAudio->uiSelectedDeviceID,
                               (LPWAVEFORMATEX)pWaveFormat,
-                              (DWORD)WaveOutCallbackRoutine,
+                              WaveOutCallbackRoutine,
                               (DWORD)pPlayAudio,
                               dwOpenFlags ))
 #endif

